daemon/system-info.sh

```diff
@@ -466,16 +466,15 @@ if [ "${VIRTUALIZATION}" != "none" ] && command -v curl > /dev/null 2>&1; then
       fi
     fi
 
-    # TODO: needs to be tested in Microsoft Azure
     # Try Azure IMDS
-    # if [ "${CLOUD_TYPE}" = "unknown" ]; then
-    #   AZURE_IMDS_DATA="$(curl --fail -s -m 5 -H "Metadata: true" --noproxy "*" "http://169.254.169.254/metadata/instance?version=2021-10-01")"
-    #   if [ -n "${AZURE_IMDS_DATA}" ]; then
-    #     CLOUD_TYPE="Azure"
-    #     CLOUD_INSTANCE_TYPE="$(curl --fail -s -m 5 -H "Metadata: true" --noproxy "*" "http://169.254.169.254/metadata/instance/compute/vmSize?version=2021-10-01&format=text")"
-    #     CLOUD_INSTANCE_REGION="$(curl --fail -s -m 5 -H "Metadata: true" --noproxy "*" "http://169.254.169.254/metadata/instance/compute/location?version=2021-10-01&format=text")"
-    #   fi
-    # fi
+    if [ "${CLOUD_TYPE}" = "unknown" ]; then
+      AZURE_IMDS_DATA="$(curl --fail -s --connect-timeout 1 -m 5 -H "Metadata: true" --noproxy "*" "http://169.254.169.254/metadata/instance?api-version=2021-10-01")"
+      if [ -n "${AZURE_IMDS_DATA}" ]; then
+        CLOUD_TYPE="Azure"
+        CLOUD_INSTANCE_TYPE="$(curl --fail -s --connect-timeout 1 -m 5 -H "Metadata: true" --noproxy "*" "http://169.254.169.254/metadata/instance/compute/vmSize?api-version=2021-10-01&format=text")"
+        CLOUD_INSTANCE_REGION="$(curl --fail -s --connect-timeout 1 -m 5 -H "Metadata: true" --noproxy "*" "http://169.254.169.254/metadata/instance/compute/location?api-version=2021-10-01&format=text")"
```

Comment:
```suggestion
    if [ "${CLOUD_TYPE}" = "unknown" ]; then
      AZURE_IMDS_DATA="$(curl --fail -s --connect-timeout 1 -m 3 -H "Metadata: true" --noproxy "*" "http://169.254.169.254/metadata/instance?api-version=2021-10-01")"
      if [ -n "${AZURE_IMDS_DATA}" ]; then
        CLOUD_TYPE="Azure"
        CLOUD_INSTANCE_TYPE="$(curl --fail -s --connect-timeout 1 -m 3 -H "Metadata: true" --noproxy "*" "http://169.254.169.254/metadata/instance/compute/vmSize?api-version=2021-10-01&format=text")"
        CLOUD_INSTANCE_REGION="$(curl --fail -s --connect-timeout 1 -m 3 -H "Metadata: true" --noproxy "*" "http://169.254.169.254/metadata/instance/compute/location?api-version=2021-10-01&format=text")"
```

Just to align with GPC/AWS curl calls